Guide/Constructive/Bonetrousle/solution.java

```diff
@@ -0,0 +1,64 @@
+import java.io.*;
+import java.util.Scanner;
+
+/**
+ * Created by koepk on 10/3/2016.
+ */
+public class Bonetrousle
+{
+    public static void main(String[] args) throws IOException
```

Comment:
Could you break this code up to make it cleaner? Split it into functions a little?